File: impl/typescript/src/main/ts/liff.ts
import { AUTHOR, ENTRIES, TITLE } from "./dictionary-generated";
import { Dictionary } from "./dictionary";
import type { Entry, Outcome, Request } from "./model";

export { AUTHOR, TITLE };
export * from "./dictionary";
export * from "./model";
export { normalize, normalizeGlob } from "./normalize";

export const DEFAULT_DICTIONARY = new Dictionary(ENTRIES);

export function entries(): readonly Entry[] {
    return DEFAULT_DICTIONARY.entries();
}

export function resolve(request: Request): Outcome {
    return DEFAULT_DICTIONARY.resolve(request);
}

File: impl/java/src/main/java/liff/core/Liff.java
package liff.core;

import java.util.List;

/** Process-wide access to the generated Meaning of Liff dictionary. */
public final class Liff {
    public static final String TITLE = GeneratedDictionary.TITLE;
    public static final String AUTHOR = GeneratedDictionary.AUTHOR;

    private static final Dictionary DICTIONARY = new Dictionary(GeneratedDictionary.ENTRIES);

    private Liff() {}

    public static Dictionary dictionary() {
        return DICTIONARY;
    }

    public static List<Entry> entries() {
        return DICTIONARY.entries();
    }

    public static Outcome resolve(Request request) {
        return DICTIONARY.resolve(request);
    }
}
